queries/flights.py

```diff
@@ -0,0 +1,543 @@
+
+    from_date: datetime
+    to_date: datetime
+    days: int = result.get("day_count", 5)  # Check 5 days into future by default
+    from_date = result.get("from_date", datetime.now(timezone.utc))
+    to_date = result.get("to_date", datetime.now(timezone.utc) + timedelta(days=days))
+
+    # Normalize airport/city names
+    airport = _LOCATION_ABBREV_MAP.get(airport, airport)
+    airport = NounPhrase(airport).nominative or airport
+
+    api_airport = _LOCATION_ABBREV_MAP.get(api_airport, api_airport)
+    api_airport = NounPhrase(api_airport).nominative or api_airport
+
+    # Translate Icelandic airport to its IATA code
+    iata_code: str = _AIRPORT_TO_IATA_MAP.get(api_airport, api_airport)
+
+    # TODO: Currently module only fetches one flight,
+    # modifications to the grammar could allow fetching of more flights at once
+    flight_count: int = result.get("flight_count", 1)
+
+    flight_data: FlightList
+    # Check first if function result in cache, else fetch data from API
+    if departing in _FLIGHT_CACHE:
+        flight_data = _FLIGHT_CACHE[departing]
+    else:
+        flight_data = _fetch_flight_data(from_date, to_date, iata_code, departing)
+
+    flight_data = _filter_flight_data(flight_data, airport, api_airport, flight_count)
+
+    answ: Dict[str, str] = dict()
+    if len(flight_data) > 0:
+        # (Format month names in Icelandic)
+        with changedlocale(category="LC_TIME"):
+            answ = _format_flight_answer(flight_data)
+    else:
+        to_airp: str
+        from_airp: str
+        if departing:
+            to_airp, from_airp = airport, api_airport
+        else:
+            from_airp, to_airp = airport, api_airport
+
+        to_airp = icelandic_city_name(capitalize_placename(to_airp))
+        from_airp = icelandic_city_name(capitalize_placename(from_airp))
+
+        from_airp = NounPhrase(from_airp).dative or from_airp
+        to_airp = NounPhrase(to_airp).genitive or to_airp
+
+        if from_airp == "*":
+            answ["answer"] = f"Ekkert flug fannst til {escape(to_airp)} næstu {days} daga."
+        elif to_airp == "*":
+            answ["answer"] = f"Ekkert flug fannst frá {escape(from_airp)} næstu {days} daga."
+        else:
+            answ["answer"] = (
+                f"Ekkert flug fannst "
+                f"frá {escape(from_airp)} "
+                f"til {escape(to_airp)} "
+                f"næstu {days} daga."
+            )
+        answ["voice"] = answ["answer"]
+
+    return answ
+
+
+def sentence(state: QueryStateDict, result: Result) -> None:
+    """Called when sentence processing is complete."""
+    q: Query = state["query"]
+    if (
+        "qtype" in result
+        and result["qtype"] == _FLIGHTS_QTYPE
+        and isinstance(result.get("departure"), bool)
+    ):
+        try:
+            answ: Dict[str, str] = _process_result(result)
+            q.set_qtype(_FLIGHTS_QTYPE)
+            q.set_answer(answ, answ["answer"], answ["voice"])
```

Comment:
You might want to set the source for flight information, (e.g. q.set_source("Isavia"))

Reply:
Added source, thanks!